File: Kapitel14/14.5/Konto/Sparkonto.cpp
/* Programmname: Konto
 * Fachkonzept-Klasse: Sparkonto
 * Aufgabe: Sparkonten dürfen nicht negativ werden
 * Dateiname: Sparkonto.cpp
 */
#include "Sparkonto.h"
// Konstruktor
Sparkonto::Sparkonto(int nummer, double ersteZahlung) : Konto(nummer, ersteZahlung)
{
}

// Redefinierte Operation
void Sparkonto::buchen(double betrag)
{
    if (getKontostand() + betrag >= 0)
        // Operation buchen der Oberklasse aufrufen
        Konto::buchen(betrag);
}

File: Kapitel15/15.4/Konto/Sparkonto.cs
/* Programmname: Konto und Sparkonto
* Fachkonzept-Klasse: Sparkonto
* Aufgabe: Verwalten von Sparkonten
* Restriktion: Sparkonten dürfen nicht negativ werden
*/
public class Sparkonto : Konto
{
    // Konstruktor
    public Sparkonto(int kontonr, double ersteZahlung):
        base(kontonr, 0.0)
    {
        buchen(ersteZahlung);
    }
    
    //Redefinierte Operation
    public override void buchen(double Betrag)
    {
        //geerbte Operation getKontostand der Oberklasse
        if (getKontostand() + Betrag >= 0)
            base.buchen(Betrag);
        //Operation buchen der Oberklasse aufrufen
    }
}